### PR TITLE
Add color emoji support on mac.

### PR DESCRIPTION
**Pull request by glennw**
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1754)

### CODE REVIEW

**Comment by glennw** [comment]
emoji

cc @Gankro The blur is not quite the same as in Gecko - but I think that's more an issue with the blur than the glyph. Nonetheless, it's better than it was.

**Comment by Gankro** [comment]
screen shot 2017-09-26 at 10 50 45 am
screen shot 2017-09-26 at 10 50 57 am
screen shot 2017-09-26 at 10 51 18 am
screen shot 2017-09-26 at 10 52 26 am

Lookin' good!

**Comment by Gankro** [comment]
Fixes #1749 and #1747 (except we still don't make them transparent with `color: rgba(0,0,0,0.5)`)

**Comment by mstange** [comment]
https://dbaron.org/log/20110225-blur-radius may provide insight.

File: webrender/res/ps_text_run.glsl [hunk 1]
```diff
@@ -70,15 +70,14 @@ void main(void) {
...
 #else
-    float alpha = texture(sColor0, tc).a;
+    vec4 color = texture(sColor0, tc) * vColor;
 #ifdef WR_FEATURE_TRANSFORM
     float a = 0.0;
     init_transform_fs(vLocalPos, a);
-    alpha *= a;
+    color.a *= a;
 #endif
-    vec4 color = vColor;
```
Comment (kvark) [inline webrender/res/ps_text_run.glsl]: This code path used to ignore `sColor0` texture fetch, and now it does not. Is this change intentional?

File: webrender/src/platform/macos/font.rs [hunk 2]
```diff
@@ -422,6 +422,14 @@ impl FontContext {
         }
     }
 
+    pub fn is_bitmap_font(&mut self, font_key: FontKey) -> bool {
+        self.get_ct_font(font_key, Au(16 * 60), &[])
+            .map_or(false, |ref ct_font| {
```
Comment (kvark) [inline webrender/src/platform/macos/font.rs]: nit: a simple match would look cleaner

Comment (glennw) [reply webrender/src/platform/macos/font.rs]: Fixed

File: webrender/res/ps_text_run.glsl (hunk 1 above)
Comment (glennw) [reply webrender/res/ps_text_run.glsl]: It used to fetch just the ```a``` component for non-subpixel fonts. Now, it fetches and scales the color by the entire ```rgba``` of the texture. This is necessary to support full color glyphs. Normal alpha glyphs that go through this path always have ```(1,1,1)``` in their ```rgb``` components, so it all works the same for those fonts (I verified this visually on all platforms).

**Comment by glennw** [comment]
@kvark Thanks - fixed and comments addressed.

**Comment by mbrubeck** [comment]
@bors-servo r-

testing

**Comment by glennw** [comment]
Merging manually due to bors issues.

File: webrender_api/src/font.rs [hunk 3]
```diff
@@ -93,6 +93,7 @@ pub enum FontRenderMode {
     Mono = 0,
     Alpha,
     Subpixel,
+    Bitmap,
```
Comment (mstange) [inline webrender_api/src/font.rs]: Would `Color` be a better name? When I read the word `Bitmap` I think of a monochrome pixel font. @jrmuizel felt the same when I asked him about it.
